src/ui/tool.ts

```diff
@@ -318,15 +318,15 @@ export class GlobalToolBinder extends RefCounted {
     this.changed.dispatch();
   }
 
-  activate(key: string): Borrowed<Tool> | undefined {
-    const tool = this.get(key);
+  activate(key: string, tool?: Tool<object>): Borrowed<Tool> | undefined {
+    tool = tool || this.get(key);
```

Comment:
Please use `tool ?? this.get(key)` instead --- not necessary here but avoids thinking about javascript truthyness complications